src/lib/tsc.ts

```diff
@@ -29,7 +29,8 @@ export function createTSProgram(
     rootNames: layout.schemaModules.concat(layout.app.exists ? [layout.app.path] : []),
     options: {
       ...compilerCacheOptions,
-      ...layout.tsConfigJson.options,
+      ...layout.tsConfig.content.options,
+      outDir: layout.buildOutput,
```

Comment:
The actual fix is here. The start module was written in the right build output, but ts was emitting to `outDir`